tests/draft2019-09/prototype-safe.json

```diff
@@ -0,0 +1,624 @@
+        "comment": "This is a baseline check to validate that other properties are treated the same and are not special compared to it",
+        "schema": {
+            "properties": { "x": { "type": "object" } }
+        },
+        "tests": [
+            { "description": "Valid on numbers", "data": 0, "valid": true },
+            { "description": "Valid on arrays", "data": [], "valid": true },
+            { "description": "Valid if not present", "data": {}, "valid": true },
+            { "description": "Valid if correct", "data": { "x": {} }, "valid": true },
+            { "description": "Invalid if incorrect (string)", "data": { "x": "foo" }, "valid": false },
+            { "description": "Invalid if incorrect (empty string)", "data": { "x": "" }, "valid": false },
+            { "description": "Invalid if incorrect (boolean true)", "data": { "x": true }, "valid": false },
+            { "description": "Invalid if incorrect (boolean false)", "data": { "x": false }, "valid": false },
+            { "description": "Invalid if incorrect (array)", "data": { "x": [42] }, "valid": false },
+            { "description": "Invalid if incorrect (empty array)", "data": { "x": [] }, "valid": false },
+            { "description": "Invalid if incorrect (zero number)", "data": { "x": 0 }, "valid": false },
+            { "description": "Invalid if incorrect (number)", "data": { "x": 42 }, "valid": false }
+        ]
+    },
+    {
+        "description": "Does not see elements non existing on the object: 'x' via required",
+        "comment": "This is a baseline check to validate that other properties are treated the same and are not special compared to it",
+        "schema": {
+            "required": ["x"]
+        },
+        "tests": [
+            { "description": "Valid on numbers", "data": 0, "valid": true },
+            { "description": "Valid on arrays", "data": [], "valid": true },
+            { "description": "Invalid if not present", "data": {}, "valid": false },
+            { "description": "Valid if present (string)", "data": { "x": "foo" }, "valid": true },
+            { "description": "Valid if present (empty string)", "data": { "x": "" }, "valid": true },
+            { "description": "Valid if present (boolean true)", "data": { "x": true }, "valid": true },
+            { "description": "Valid if present (boolean false)", "data": { "x": false }, "valid": true },
+            { "description": "Valid if present (array)", "data": { "x": [42] }, "valid": true },
+            { "description": "Valid if present (empty array)", "data": { "x": [] }, "valid": true },
+            { "description": "Valid if present (object)", "data": { "x": {} }, "valid": true },
+            { "description": "Valid if present (zero number)", "data": { "x": 0 }, "valid": true },
+            { "description": "Valid if present (number)", "data": { "x": 42 }, "valid": true }
+        ]
+    },
```

Comment:
What is this testing?

Reply:
The `x` property test is a baseline test, as compared to everything below. It's needed only as a reference against other tests, which have a similar structure, but check properties that might end up special handled.

Reply:
For example, `revalidator` fails on both `toString via required` and  `toString as number`.

It also fails on `x via required` which is indicating that it has a general `required` problem which is not necessary related to prototype safety.

But it doesn't fail on `x as number` while failing on `toString as number`, which indicates that it has a prototype safety problem.